### PR TITLE
8236839: System menubar removed when other menubars are created or modified

### PR DESCRIPTION
**Pull request by jkaving**
This pull request fixes the sceneProperty listener in `MenuBarSkin` so that we leave the
current system menubar alone when other menubars are changed.

It also adds a test case that reproduces the problem before the fix.

## Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

## Issue
[JDK-8236839](https://bugs.openjdk.java.net/browse/JDK-8236839): System menubar removed when other menubars are created or modified

## Approvers
 * Kevin Rushforth ([kcr](@kevinrushforth) - **Reviewer**)
 * Ajit Ghaisas ([aghaisas](@aghaisas) - **Reviewer**)

### CODE REVIEW

**Comment by jkaving** [comment]
I just realised that I did not make the commit with the same email address that I used on the OCA (but both addresses are linked to my GitHub account).
Will this be a problem?
Should I redo the commit with the correct email address and do a force push to my branch?

**Comment by kevinrushforth** [comment]
No, it's not a problem. The commit email address doesn't matter as long as your OCA can be validated.

**Comment by kevinrushforth** [comment]
@aghaisas can be the second reviewer.

File: modules/javafx.controls/src/test/java/test/javafx/scene/control/skin/MenuBarSkinTest.java [hunk 1]
```diff
@@ -26,9 +26,11 @@
 package test.javafx.scene.control.skin;
 
 import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
 
 import com.sun.javafx.menu.MenuBase;
 import com.sun.javafx.stage.WindowHelper;
+import javafx.scene.control.MenuItem;
```
Comment (kevinrushforth) [inline modules/javafx.controls/src/test/java/test/javafx/scene/control/skin/MenuBarSkinTest.java]: Minor: can you move this next to the other `javafx.scene.control` imports? (I know they aren't all sorted, but most are)

Comment (jkaving) [reply modules/javafx.controls/src/test/java/test/javafx/scene/control/skin/MenuBarSkinTest.java]: Of, course - I've added a commit that moves the import statement.

Comment (jkaving) [reply modules/javafx.controls/src/test/java/test/javafx/scene/control/skin/MenuBarSkinTest.java]: > I just realised that I did not make the commit with the same email address that I used on the OCA (but both addresses are linked to my GitHub account).

And I now did the mistake of committing using the email address on the OCA, which is different than the one used in the previous commit.
`jcheck` complained about the differing email addresses so I have force pushed a commit using the same address as before.

Comment (kevinrushforth) [reply modules/javafx.controls/src/test/java/test/javafx/scene/control/skin/MenuBarSkinTest.java]: Oh, I see. I was going to ask you about the force-push, but this explains it.

**Pull request closed without merging**